### PR TITLE
Add promise behavior

### PR DESCRIPTION
**Pull request by alancnet**
Adds promise behavior, but preserves existing interface. Only transforms into a promise interface if you call `.then(...)`

Needs version bump

### CODE REVIEW

**Comment by richorama** [comment]
Nice! Give me 24hrs to review and test the code

**Comment by alancnet** [comment]
Added documentation to [readme.md](https://github.com/azure-contrib/node-azure-search/pull/25/commits/225acd144039cb35b878f5d113bf2c18a2239ec3)

**Comment by alancnet** [comment]
Added documentation to [readme.md](https://github.com/alancnet/node-azure-search/compare/8ac4532808de9d5a94efd882662e6d9e335d0e98...alancnet:eeea4bd798688ebe05470313cb1dae9baf9821b9)

File: index.js [hunk 1]
```diff
@@ -364,8 +364,30 @@ module.exports = function (options) {
         if (data && data.error) return cb(data.error, null, data)
         cb(null, null, data)
       })
+    },
+    then: function (res, rej) {
+      var ret = {}
+      var self = this
+      Object.keys(this).forEach(function (key) {
+        if (key !== 'then') {
```
Comment (richorama) [inline index.js]: Should the 'catch' function be ignored also?

Comment (alancnet) [inline index.js]: Indeed!

**Comment by alancnet** [comment]
@richorama have you reviewed since the last commit?

**Comment by richorama** [comment]
Ah, I hadn't seen that. Thanks for brining it to my attention!

**Comment by richorama** [comment]
Thanks for your contribution. I'll publish to npm when I get a chance.

**Pull request merged**